Python/bltinmodule.c

```diff
@@ -1976,9 +1976,15 @@ builtin_input_impl(PyObject *module, PyObject *prompt)
             promptstr = PyBytes_AS_STRING(po);
         }
         else {
-            po = NULL;
+            po = PyBytes_FromString("");
             promptstr = "";
         }
+        if (PyBytes_GET_SIZE(po) != strlen(promptstr)) {
+                PyErr_SetString(PyExc_ValueError,
+                                "input prompt has NULL character");
```

Comment:
How about `"input: prompt string cannot contain null characters"`? "null" is more common in existing error messages, but "NUL" with one L is also used, since that's the ASCII name of character. `NULL` refers to the C pointer constant.